sum_types_generator/lib/src/sum_type_spec.dart

```diff
@@ -29,24 +29,36 @@ class CaseSpec {
   const CaseSpec({
     required this.name,
     required this.type,
+    required this.namedPayload,
+    required this.positionalPayload,
+    required this.parameterName,
   });
 
   @override
   bool operator ==(dynamic other) =>
       other.runtimeType == runtimeType &&
       other.name == name &&
-      other.type == type;
+      other.type == type &&
+      other.namedPayload == namedPayload &&
+      other.positionalPayload == positionalPayload &&
+      other.parameterName == parameterName;
 
   @override
   int get hashCode {
     var result = 17;
     result = 37 * result + name.hashCode;
     result = 37 * result + type.hashCode;
+    result = 37 * result + namedPayload.hashCode;
+    result = 37 * result + positionalPayload.hashCode;
+    result = 37 * result + parameterName.hashCode;
     return result;
   }
 
   final String name;
   final CaseTypeSpec type;
+  final bool namedPayload;
+  final bool positionalPayload;
+  final String parameterName;
```

Comment:
This is not wrong, but has a flow: as defined now, an instance of `CaseSpec` can have both `namedPayload` and `positionalPayload` set, which does not encode a valid configuration.

This is a typical problem with having a bunch of (dependent) boolean flags: they can potentially be set in a way that doesn't make sense.

When possible, it's better to have your types be only capable of encoding the valid configurations.

What we actually need to encode here is either "it's a positional parameter" or "it's a named parameter", two options. But two boolean flags can have four (2^2) possible configurations, right? So, there are two unneeded options and it's not exactly clear what the program should do, if one of those unneeded options appears.

What options do we have in this case? I can think of at least two:

1. have a single flag, e.g. `isParameterNamed`; when not set, assume it's positional
2. have a single property, e.g. `parameterStyle`, of type `enum ParameterStyle { positional, named }`

The first approach is minimalistic and could work here, but it has an _implicit_ part: the flag is called "named" and if it's not set, then what? Unnamed? "Unnamed" is not exactly the same as "positional", so this part is a bit fishy.

The second approach allows to explicitly name both styles, there is nothing implicit (nothing to guess, no arcane knowledge required). It's also scalable: if there would be three options, you'd be able to explicitly define exactly three. This is why I'm leaning towards the second option here.